metadata-ingestion/docs/sources/dataplex/dataplex_pre.md

```diff
@@ -0,0 +1,401 @@
+| :------------------------ | :------------------------------------------------------------------------------ | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Entry (Universal Catalog) | [`Dataset`](https://docs.datahub.com/docs/generated/metamodel/entities/dataset) | Metadata from Universal Catalog. Ingested using **source platform URNs** (e.g., `bigquery`, `gcs`). Schema metadata is extracted when available.                       |
+| Entity (Lakes/Zones)      | [`Dataset`](https://docs.datahub.com/docs/generated/metamodel/entities/dataset) | Discovered table or fileset from lakes/zones. Ingested using **source platform URNs** (e.g., `bigquery`, `gcs`). Schema metadata is extracted when available.          |
+| Lake/Zone/Asset           | Custom Properties                                                               | Dataplex hierarchy information (lake, zone, asset, zone type) is preserved as **custom properties** on datasets for traceability without creating separate containers. |
+
+#### API Selection Guide
+
+**When to use Entries API** (default, `include_entries: true`):
+
+- ✅ You want to discover all BigQuery tables, Pub/Sub topics, and other Google Cloud resources
+- ✅ You need comprehensive metadata from Dataplex's centralized catalog
+- ✅ You want system-managed discovery without manual lake/zone configuration
+- ✅ **Recommended for most users**
+
+**When to use Entities API** (`include_entities: true`):
+
+- Use this if you need entity-level details specific to lakes/zones that aren't available in the Entries API
+- Provides Dataplex organizational context (lake, zone, asset metadata)
+- Can be used alongside Entries API, but see warning below
+
+**Important**: To access system-managed entry groups like `@bigquery` that contain BigQuery tables, you must use **multi-region locations** (`us`, `eu`, `asia`) via the `entries_location` config parameter. Regional locations (`us-central1`, etc.) only contain placeholder entries.
+
+#### ⚠️ Using Both APIs Together - Important Behavior
+
+When both `include_entries` and `include_entities` are enabled and they discover the **same table** (same URN), the metadata behaves as follows:
+
+**What gets preserved:**
+
+- ✅ Schema metadata (from Entries API - most authoritative)
+- ✅ Entry-specific custom properties (`dataplex_entry_id`, `dataplex_entry_group`, `dataplex_fully_qualified_name`, etc.)
+
+**What gets lost:**
+
+- ❌ Entity-specific custom properties (`dataplex_lake`, `dataplex_zone`, `dataplex_zone_type`, `data_path`, `system`, `format`, `asset`)
+
+**Why this happens:** DataHub replaces aspects at the aspect level. When the Entries API emits metadata for a dataset that was already processed by the Entities API, it completely replaces the `datasetProperties` aspect, which contains all custom properties.
```

Comment:
maybe not in the scope of this PR but instead of the source ingestor emitting conflicting aspects that overwrite each other, we could make a merge of those aspects on the source ingestor side itself

Reply:
@sgomezvillamor I considered that approach but I was concerned about the performance implications of holding everything in memory from the first api until merged. I was thinking the ideal approach would be to use patching but I didn't want to add that complexity yet. Using structured properties also seemed like unnecessary complexity.

Should I ignore the potentially memory implications or am I misunderstanding them?

Reply:
I’m not sure how concerning the memory impact is, we’ll find out once we deploy this to customers.
IMO Let’s get some real-world experience before making any final decision.